### PR TITLE
Handbook: add components readme file

### LINKED ISSUE #10702: Handbook: add "Components Package Reference" md file
Currently, the file doesn't exist, but it is referenced in the handbook.

https://wordpress.org/gutenberg/handbook/components/

https://raw.githubusercontent.com/WordPress/gutenberg/master/packages/components.md

We should add one, and also add a list of available components. It would have to be manually updated, unless someone has an alternate solution.

### PR DESCRIPTION
**Pull request by davewhitley**
fixes #10702 

Part of the handbook master thread #10549 

## Description

This adds a components readme file that will display on the handbook site: https://wordpress.org/gutenberg/handbook/components/

Currently, it's blank and does not exist in this repo.

I also categorized the components into either a UI component, or a utility component. This makes it easier to find what you are looking for.

Some of these components have not been added to the manifest file — we'll need to add them there after this PR is merged. Or just update the manifest file in this PR?

https://github.com/WordPress/gutenberg/blob/master/docs/manifest.json

I'd like a review on the URL strings and whether or not I categorized the components correctly. Thanks!

### CODE REVIEW

File: docs/components/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Components Package Reference
+
+This is a library of generic WordPress components to be used for creating common UI elements shared between screens and features of the WordPress dashboard. They are published to vpm via the `@wordpress/components` package.
```
Comment (Soean) [inline docs/components/README.md]: `vpm` --> `npm`

**Comment by drw158** [comment]
I believe the individual component readmes can stay where they are, and they'll be linked the same way they are now, like https://wordpress.org/gutenberg/handbook/components/autocomplete/

which links to 

https://github.com/WordPress/gutenberg/edit/master/packages/components/src/autocomplete/README.md

I don't have a strong opinion on whether that should change or not though.

**Comment by mtias** [comment]
> Is the plan that individual readmes from each package will get merged into the docs folder or will we need to write separate documentation for the components?

I want to keep the documentation for each component next to the code. That'll ensure that people consuming the packages can get the relevant readmes and guidelines.

Our /docs can aggregate them if needed.

**Comment by youknowriad** [comment]
We already aggregate the components with our "docs build tool". I'm thinking we should consolidate the work on this PR with what we already have.

**Review by mtias — APPROVED** [review]
This looks like a great start

**Comment by mtias** [comment]
> By this I mean. Automatically generate this file.

We might want to categorize and group them manually, though.

**Comment by youknowriad** [comment]
> We might want to categorize and group them manually, though.

I'm afraid we'll forget to update this list. Can't we add a comment tag or something?

**Comment by youknowriad** [comment]
That said, I'm fine with manual process first.

**Comment by mtias** [comment]
> Can't we add a comment tag or something?

Maybe. But I think we should first see what we want as organization (should we have a group for Button that includes IconButton ButtonGroup etc?). Once that is more settled we can look at automating it.

**Comment by drw158** [comment]
Good discussion. I wasn't sure what to do with `HigherOrder`. It seems to be a group of components, but it's listed as an individual component. Also, I don't know what it's used for :)

Automated process would be great. I can look into what kind of organization we could have.

**Comment by drw158** [comment]
I just double-checked the list and I found a couple that I missed. This is ready for review.

I still haven't updated the manifest file. I think I may need to update that file. I can do that after the PR is merged though.

**Comment by drw158** [comment]
Here's my suggestion for categorizing the components:

- **Buttons**
  - Button
  - IconButton
  - ButtonGroup
- **Inputs and Controls**
  - TextControl
  - TextareaControl
  - SelectControl
  - TreeSelect
  - FormToggle
  - ToggleControll
  - RadioControl
  - CheckboxControl
  - RangeControl
  - BaseControl
  - DateTimePicker
  - FormTokenField
- **Menus**
  - MenuGroup
  - Dropdown
  - DropdownMenu
  - MenuItem
  - MenuItemsChoice

All the rest can remain uncategorized. 

Ideally we could have many components on the same page, for example, the menu components. But, I'm not sure how that'd work with the separate readme files.

**Comment by ajitbohra** [comment]
> Some of these components have not been added to the manifest file — we'll need to add them there after this PR is merged. Or just update the manifest file in this PR?
> 
> [/docs/manifest.json@`master`](https://github.com/WordPress/gutenberg/blob/master/docs/manifest.json)

`manifest.json` is auto generated with `npm run docs:build`

> I still haven't updated the manifest file. I think I may need to update that file. I can do that after the PR is merged though.

You can update the `manifest` in this PR, entry for components parent needs to be added to `root-manifest.js` https://github.com/WordPress/gutenberg/blob/master/docs/root-manifest.json

**Comment by gziolo** [comment]
The script for generating manifest is here: https://github.com/WordPress/gutenberg/blob/master/docs/tool/index.js - all code is in the same folder. It basically scans various places in the repository and using `glob` discover useful `README` documents to expose in Gutenberg Handbook. They are grouped together using very random rules :)

**Comment by gziolo** [comment]
@chrisvanpatten - with the recent docs reorganization it seems like out of date. Can you help to either merge it or close?

**Review by gziolo — CHANGES_REQUESTED** [review]
It is out of date.

**Comment by gziolo** [comment]
Thanks for your time spent on this PR. I agree that at some point we should add some organization to the components package and add some subfolders. I don't think it's so pressing as of today.

Regarding keeping the list of components it might be hard to maintain manually so I would prefer we spend the time to automate it in a similar fashion we did it for the handbook's manifest.

I'm closing this PR but we can use this work as a template to automatically update `@wordpress/components` README file with the list of components to have it included on this page: https://wordpress.org/gutenberg/handbook/designers-developers/developers/components/.

**Pull request closed without merging**